### ISSUE TITLE
xmldom related cpu 100%

### ISSUE THREAD

**Issue by jseagull** [mattiasw/ExifReader#240]
### Description

I found the image that cause cpu 100%, it has a 12.7 MB XML which will take hours to parse.

xmldom parses the whole xml very quickly, but when started from the middle of it, it will take hours. 

[exif.jpeg.zip](https://github.com/mattiasw/ExifReader/files/11859426/exif.jpeg.zip)

https://github.com/xmldom/xmldom/issues/501

**Comment by mattiasw** [comment]
Hi! How do you mean start from the middle?

**Comment by jseagull** [comment]
```
               <rdf:li>048E56CC83C09EBF3FC9303BF6DCB918</rdf:li>
               <rdf:li>048F302A78A1F5910BA1912CCB0C07D9</rdf:li>
               <rdf:li>048FA2A7DE1F4F1FDC5C8AE511085C66</rdf:li>
               <����http://ns.adobe.com/xap/1.0/rdf:li>0491CCFD8BE0CD8AAEA1849CA9C54F1B</rdf:li>
               <rdf:li>049280C90EF43C129D222AAEF8187174</rdf:li>
               <rdf:li>04928DB0EEE1FC151868609020E353D4</rdf:li>
               <rdf:li>0493DEC43BC4176B264877D88A7CD05F</rdf:li>
```


The whole xml  is separated into two parts. The first part is from beginning to here


```
               <rdf:li>048E56CC83C09EBF3FC9303BF6DCB918</rdf:li>
               <rdf:li>048F302A78A1F5910BA1912CCB0C07D9</rdf:li>
               <rdf:li>048FA2A7DE1F4F1FDC5C8AE511085C66</rdf:li>
               <
```

The second part is from here to the end of xml.


```
rdf:li>0491CCFD8BE0CD8AAEA1849CA9C54F1B</rdf:li>
               <rdf:li>049280C90EF43C129D222AAEF8187174</rdf:li>
               <rdf:li>04928DB0EEE1FC151868609020E353D4</rdf:li>
               <rdf:li>0493DEC43BC4176B264877D88A7CD05F</rdf:li>
```

**Comment by karfau** [comment]
You say it takes hours to parse. Do you say that because you tried to let it run for hours and it finished to parse it at some point?
I would rather have expected that it runs into an endless loop which never exits.

I would really like to understand the the reason for passing incomplete XML to xmldom, since it's not something that I would have expected to ever work (reliably).

and as suggested [here](https://github.com/xmldom/xmldom/issues/501#issuecomment-1606655881), I can only recommend the add an `errorHandler` to the DOMParser constructor to stop parsing any XML that is not well-formed:

Even when parsing the whole document, xmldom only reports "error"s, which doesn't stop it from continuing parsing. I can only recommend to add an `errorHandler` that throws a `ParseError` on every reported `error` or even `warning`, especially when the documents are coming from user input:
in currently available versions: 
```
const domParser = new DOMParser({errorHandler:(level) => {
  if (level === 'error') {
    throw new ParseError('stop parsing due to reported error');
  }
}});
```
starting with 0.9.0-beta.9 (not released yet)
```
const domParser = new DOMParser({onError:onErrorStopParsing});
```

**Comment by jseagull** [comment]
I am using exifreader on my website to extract EXIF information from user-uploaded images. However, sometimes the CPU usage reaches 100% for hours. Upon examining the CPU profile, I noticed that most of the CPU usage is attributed to XMLDOM. Exifreader reads XMP information from the image, but I did not find any endless loops in the code. 
This XML file is extracted from one of the user-uploaded images as I posted above.
